camkes/templates/component.common.c

```diff
@@ -42,6 +42,18 @@
 
 /*? macros.show_includes(me.type.includes) ?*/
 
+/*- if 'affinity' in configuration[me.name].keys() -*/
+#if(/*? configuration[me.name].get('affinity') ?*/ >= CONFIG_MAX_NUM_NODES)
+#error "Invalid Affinity Value of /*? configuration[me.name].get('affinity') ?*/"
+#endif
+/*- endif -*/
+
+/*- if 'num_vcpus' in configuration[me.name].keys() -*/
```

Comment:
Similarly to [my other comment](https://github.com/seL4/camkes-tool/pull/122#issuecomment-1323351873), a check that's specific to the VM_Arm component type can be implemented using a template that's specific to that component.

Reply:
Dropped the num_vcpus check from this MR, will create another one in camkes-vm